js/components/Message/BodyPieces.js

```diff
@@ -8,10 +8,15 @@ BodyPieces.propTypes = {
 function BodyPieces(props) {
   //@TODO add support for styles.
 
-  let pieces = props.pieces.map(function(piece) {
+  let pieces = props.pieces.map(function(piece, i) {
     const separator = " ";
+    // Body pieces don't ever change order so we can use the index
```

Comment:
I think this should be fine based on my reading of http://facebook.github.io/react/docs/multiple-components.html#dynamic-children